crates/storage/src/ops/l1tx_broadcast.rs

```diff
@@ -24,73 +20,13 @@ impl<D: BroadcastDatabase + Sync + Send + 'static> Context<D> {
     }
 }
 
-inst_ops! {
-    (BroadcastDbOps, Context<D: BroadcastDatabase>) {
+inst_ops_auto! {
+    (l1_broadcast_db, BroadcastDbOps, Context<D: BroadcastDatabase>) {
         get_tx_entry(idx: u64) => Option<L1TxEntry>;
         get_tx_entry_by_id(id: Buf32) => Option<L1TxEntry>;
-        get_tx_status(id: Buf32) => Option<L1TxStatus>;
         get_txid(idx: u64) => Option<Buf32>;
         get_next_tx_idx() => u64;
         put_tx_entry(id: Buf32, entry: L1TxEntry) => Option<u64>;
         put_tx_entry_by_idx(idx: u64, entry: L1TxEntry) => ();
     }
 }
-
-fn get_tx_entry<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-    idx: u64,
-) -> DbResult<Option<L1TxEntry>> {
-    let bcast_db = context.db.l1_broadcast_db();
-    bcast_db.get_tx_entry(idx)
-}
-
-fn get_tx_entry_by_id<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-    id: Buf32,
-) -> DbResult<Option<L1TxEntry>> {
-    let bcast_db = context.db.l1_broadcast_db();
-    bcast_db.get_tx_entry_by_id(id)
-}
-
-fn get_txid<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-    idx: u64,
-) -> DbResult<Option<Buf32>> {
-    let bcast_db = context.db.l1_broadcast_db();
-    bcast_db.get_txid(idx)
-}
-
-fn get_tx_status<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-    id: Buf32,
-) -> DbResult<Option<L1TxStatus>> {
-    let bcast_db = context.db.l1_broadcast_db();
-    Ok(bcast_db.get_tx_entry_by_id(id)?.map(|entry| entry.status))
-}
-
-fn get_next_tx_idx<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-) -> DbResult<u64> {
-    let bcast_db = context.db.l1_broadcast_db();
-    bcast_db.get_next_tx_idx()
-}
-
-fn put_tx_entry<D: BroadcastDatabase + Sync + Send + 'static>(
-    context: &Context<D>,
-    txid: Buf32,
-    entry: L1TxEntry,
-) -> DbResult<Option<u64>> {
-    trace!(%txid, "insert_new_tx_entry");
-    assert!(entry.try_to_tx().is_ok(), "invalid tx entry {entry:?}");
```

Comment:
This is an example of where it might be useful/necessary to have this extra indirection instead of making the system completely all-encompassing but closed-off.

Reply:
This can and possibly should be done on the manager.

Reply:
If we did it there then we'd need to duplicate these checks.  You can make *an* argument that if we really are leaning into it being a thin wrapper then this wouldn't belong, but then we'd just have to deal with that.  In this context also know we're always interacting with the database in a blocking fashion.

Reply:
> If we did it there then we'd need to duplicate these checks.

I am not sure how, because if ops is just an abstraction over db, then it's not the duty of the abstraction to do the checks, it should be somewhere upwards(manager/handler), isn't it?

Reply:
The point is that it allows us flexibility to add these here if we want to, separation of concerns and all that.